src/Lean/PrettyPrinter/Formatter.lean

```diff
@@ -514,14 +514,28 @@ instance : Coe (Formatter → Formatter → Formatter) FormatterAliasValue := {
 end Formatter
 open Formatter
 
+register_builtin_option pp.oneline : Bool := {
+  defValue := false
+  group    := "pp"
+  descr    := "(pretty printer) elide all but first line of pretty printer output"
+}
+
 def format (formatter : Formatter) (stx : Syntax) : CoreM Format := do
   trace[PrettyPrinter.format.input] "{Std.format stx}"
   let options ← getOptions
   let table ← Parser.builtinTokenTable.get
   catchInternalId backtrackExceptionId
     (do
-      let (_, st) ← (concat formatter { table := table, options := options }).run { stxTrav := Syntax.Traverser.fromSyntax stx };
-      pure $ Format.fill $ st.stack.get! 0)
+      let (_, st) ← (concat formatter { table, options }).run { stxTrav := .fromSyntax stx }
+      let mut f := st.stack[0]!
+      if (pp.oneline.get? options).getD (trace.profiler.get options) then
```

Comment:
Please don't couple `pp.oneline` to `trace.profiler`.  The `trace.profiler` option sounds super useful, I don't want it to cut off half the trace message and make them noninteractive.

Reply:
I expected this to be the more useful default for my use cases, but I don't mind avoiding this option dependency